src/modules/launcher/Plugins/Microsoft.Plugin.Program/Storage/PackageRepository.cs

```diff
@@ -0,0 +1,86 @@
+    /// This repository will also monitor for changes to the PackageCatelog and update the repository accordingly
+    /// </summary>
+    internal class PackageRepository : ListRepository<UWP.Application>, IRepository<UWP.Application>, IProgramRepository
+    {
+        IPackageCatalog _packageCatalog;
+        public PackageRepository(IPackageCatalog packageCatalog, IStorage<IList<UWP.Application>> storage) : base(storage)
+        {
+            _packageCatalog = packageCatalog ?? throw new ArgumentNullException("packageCatalog", "PackageRepository expects an interface to be able to subscribe to package events");
+            _packageCatalog.PackageInstalling += OnPackageInstalling;
+            _packageCatalog.PackageUninstalling += OnPackageUninstalling;
+        }
+
+        public void OnPackageInstalling(PackageCatalog p, PackageInstallingEventArgs args)
+        {
+            if (args.IsComplete)
+            {
+
+                try
+                {
+                    var uwp = new UWP(args.Package);
+                    uwp.InitializeAppInfo(args.Package.InstalledLocation.Path);
+                    foreach (var app in uwp.Apps)
+                    {
+                        Add(app);
+                    }
+                }
+                //InitializeAppInfo will throw if there is no AppxManifest.xml for the package. 
+                //Note there are sometimes multiple packages per application and this doesn't necessarily mean that we haven't found the app.
+                //eg. "Could not find file 'C:\\Program Files\\WindowsApps\\Microsoft.WindowsTerminalPreview_2020.616.45.0_neutral_~_8wekyb3d8bbwe\\AppxManifest.xml'."
+
+                catch (System.IO.FileNotFoundException e)
+                {
+                    ProgramLogger.LogException($"|UWP|OnPackageInstalling|{args.Package.InstalledLocation}|{e.Message}", e);
+                }
+            }
+        }
+
+        public void OnPackageUninstalling(PackageCatalog p, PackageUninstallingEventArgs args)
+        {
+            if (args.Progress == 0)
+            {
+                //find apps associated with this package. 
+                var uwp = new UWP(args.Package);
+                var apps = _items.Where(a => a.Package.Equals(uwp)).ToArray();
+                foreach (var app in apps)
+                {
+                    Remove(app);
+                }
+            }
+        }
+
+        public void IndexPrograms()
+        {
+            var windows10 = new Version(10, 0);
+            var support = Environment.OSVersion.Version.Major >= windows10.Major;
+
+            var applications = support ? Programs.UWP.All() : new Programs.UWP.Application[] { };
+            Set(applications);
+        }
+
+        public void Save()
+        {
+            _storage.Save(_items);
+        }
+
+        public void Load()
+        {
+            var items = _storage.TryLoad(new Programs.UWP.Application[] { });

```

Comment:
It should work but would it be possible to pull from master to validate if the cache clears as expected?

Reply:
Will do

Reply:
@alekhyareddy28 Done.  I verified this by deleting the version text and seeing that it was re-written.  Let me know if this needs a more rigorous test.   Notice the different timestamps for the UWP.cache

![image](https://user-images.githubusercontent.com/56318517/86978677-464af880-c134-11ea-9df5-d07f5e2bcf08.png)

Reply:
I think that's good enough. Thank you for validating it.